spot_driver/src/spot_driver/spot_ros.py

```diff
@@ -1310,6 +1312,16 @@ def publish_feedback(self):
     def publish_allow_motion(self):
         self.motion_allowed_pub.publish(self.allow_motion)
 
+    def check_for_subscriber(self):
+        for pub in list(self.lookup.keys()):
+            if (
+                self.lookup[pub] not in self.added_tasks
+                and pub.get_num_connections() > 0
+            ):
+                self.spot_wrapper.update_image_tasks(self.lookup[pub])
+                self.added_tasks.append(self.lookup[pub])
+                print("Adding :", self.lookup[pub])
```

Comment:
```suggestion
        for pub in list(self.lookup.keys()):
            task_name = self.camera_pub_to_async_task_mapping[pub]
            if (
                task_name not in self.added_tasks
                and pub.get_num_connections() > 0
            ):
                self.spot_wrapper.update_image_tasks(task_name)
                self.active_camera_tasks.append(task_name)
                print(f"Detected subscriber for {task_name} task, adding task to publish")
```

Reusing the lookup a lot here, it's nicer to extract that to a variable.